src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketPal.Windows.cs

```diff
@@ -161,6 +162,12 @@ public static SocketError GetPeerName(SafeSocketHandle handle, byte[] buffer, re
             return errorCode == SocketError.SocketError ? GetLastSocketError() : SocketError.Success;
         }
 
+        public static unsafe SocketError GetPeerName(SafeSocketHandle handle, byte* buffer, int* nameLen)
```

Comment:
Can we just have the above method delegate to this one and delete the extra DllImport?